File: ZeroJudge/Normal/e/e273.py
"""
Problem link: https://zerojudge.tw/ShowProblem?problemid=e273 
Status:
"""

length = int(input())
n = input()
ans = n.split(" ")

# for i in range(0, n) :
#     ans[i] = int(input())

index = len(ans) - 1

for i in range(0, length - 1) :
    temp = int(ans[i]) * index
    print(temp, end = " ")
    index -= 1

print(" ")

File: 2023/ZeroJudge/Normal/e/e273.cpp
/*
 Problem link: https://zerojudge.tw/ShowProblem?problemid=e273
 Status: 
*/

#include <iostream>
#include <cstdlib>

using namespace std;

int main(void) {
    int n;
    
    cin >> n;
    int ans[n - 1];

    for (int i = 0; i < n; i++) {
        int temp;
        cin >> temp;
        ans[i] = temp;
    }
    int index = n - 1;
    for (int i = 0; i < n - 1; i++) {
        long long int temp = ans[i] * index;  
        printf("%d ", temp);
        index--;
    }

    cout << "\n";

    return 0;
}